src/cli/setup/compute_sig.rs

```diff
@@ -144,35 +186,64 @@ impl<'a> ComputeSigCommand {
             x_i: node_secret_share,
         };
 
-        let (is_positive, block_shared_keys, _local_sig) = Vss::create_local_sig_from_shares(
-            &priv_shared_keys,
-            index,
-            &shared_block_secrets,
-            &block,
-        )?;
+        let (is_positive, block_shared_keys, _local_sig) = if let Some(ref block) = block {
+            Vss::create_local_sig_from_shares_for_block(
+                &priv_shared_keys,
+                index,
+                &shared_block_secrets,
+                &block,
+            )?
+        } else if xfield != XField::None {
+            Vss::create_local_sig_from_shares_for_xfield(
+                &priv_shared_keys,
+                index,
+                &shared_block_secrets,
+                &xfield,
+            )?
+        } else {
+            return Err(Error::InvalidArgs(
+                "Either xfield or block is expected".to_string(),
+            ));
+        };
 
         let shared_secrets = vss_to_shared_secret_map(&node_vss_vec, &params);
 
         let mut signatures = BTreeMap::new();
         for (sig, public_key) in keyed_local_sigs {
             signatures.insert(SignerID { pubkey: public_key }, (sig.gamma_i, sig.e));
         }
+
+        let hash = if let Some(ref block) = block {
+            block.header.signature_hash().to_vec()
+        } else if xfield != XField::None {
+            xfield.signature_hash().unwrap().to_vec()
+        } else {
+            return Err(Error::InvalidArgs(
```

Comment:
Will this line be executed?
I think this line is unreachable, as the same check is on line 204.

Reply:
yes. this is unreachable. I'll fix it.